### PR TITLE
feat: check for accounting approval modules

### PR DESCRIPTION
**Pull request by 0xyaco**
# 🤖 Linear

Closes GRT-163

## Description
* Throws and fails the whole agent whenever there's at least one module pending approval

### CODE REVIEW

**Review by 0xnigir1 — COMMENTED** [review]
just a single comment, then is good for me 🫡

**Review by 0xnigir1 — APPROVED** [review]

File: apps/agent/src/index.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { inspect } from "util";
 import { EboActorsManager, EboProcessor } from "@ebo-agent/automated-dispute";
 import { ProtocolProvider } from "@ebo-agent/automated-dispute/dist/providers/protocolProvider.js";
+import { AccountingModules } from "@ebo-agent/automated-dispute/dist/types/prophet.js";
```
Comment (0xkenj1) [inline apps/agent/src/index.ts:4]: `dist` is pointing to the code that has been built ?

File: packages/automated-dispute/src/services/eboProcessor.ts [hunk 2]
```diff
@@ -48,6 +62,42 @@ export class EboProcessor {
...
+    private async checkAllModulesApproved() {
+        const approvedModules: Address[] =
+            await this.protocolProvider.getAccountingApprovedModules();
+
+        const summary: Record<"approved" | "notApproved", Partial<AccountingModules>> = {
+            approved: {},
+            notApproved: {},
+        };
+
+        for (const [moduleName, moduleAddress] of Object.entries(this.accountingModules)) {
```
Comment (0xkenj1) [inline packages/automated-dispute/src/services/eboProcessor.ts:79]:
```suggestion
        for (const key in this.accountingModules) {
```
You can do it this way also, just to let you know, not requesting any changes here 🤣

File: apps/agent/src/index.ts (hunk 1 above)
Comment (0xyaco) [reply apps/agent/src/index.ts]: :woozy_face:

File: packages/automated-dispute/src/services/eboProcessor.ts [hunk 3]
```diff
@@ -3,11 +3,22 @@ import { BlockNumberService } from "@ebo-agent/blocknumber";
 import { Caip2ChainId } from "@ebo-agent/blocknumber/dist/types.js";
```
[the remark this answers is not in the document]
Comment (0xyaco) [reply packages/automated-dispute/src/services/eboProcessor.ts:3]: Totally! Already did it in #49 's latest commit

**Review by 0xnigir1 — APPROVED** [review]

**Pull request merged**